Okta/OktaAuth/OktaTokenManager.swift

```diff
@@ -49,48 +48,32 @@ open class OktaTokenManager: NSObject, NSCoding {
     }
 
     open var refreshToken: String? {
-        // Return the known refreshToken
-        get {
-            guard let token = self.authState.refreshToken else { return nil }
-            return token
-        }
+        return self.authState.refreshToken
     }
+    
+    var restAPI: OktaHttpApiProtocol = OktaRestApi()
 
-    public init(authState: OIDAuthState, config: OktaAuthConfig, accessibility: CFString = kSecAttrAccessibleWhenUnlockedThisDeviceOnly) {
```

Comment:
Why config dependency was removed? I think it is logical that authState tied up with with the config that was used for this concrete auth flow. As a developer I can easily deserialize OktaTokenManager and start using it's API without even instantiating OktaConfig instance. We don't need dependencies on OktaAuth in this class

Reply:
Moreover, we probably don't need config at all, as OIDAuthState should contain all required data:
- Issuer url (at OIDAuthState.OIDAuthorizationResponse.OIDAuthorizationRequest.OIDServiceConfiguration)
- CliendID (at OIDAuthState.OIDAuthorizationResponse.OIDAuthorizationRequest)
- discoveryDictionary (at OIDAuthState.OIDAuthorizationResponse.OIDAuthorizationRequest.OIDServiceConfiguration.OIDServiceDiscovery)

Use this data as DI for token tasks and get rid of OktaAuth dependency

Reply:
Reworked.